From 27d6aa8d4acc898efa605da2460023e354898794 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Diego=20Elio=20Petten=C3=B2?= <flameeyes@flameeyes.com>
Date: Sun, 12 Apr 2020 13:48:58 +0100
Subject: [PATCH 037/134] Don't crash out as soon as hitting an unsupported
 condition.

This changes both the fatal into an error, and the loop to report _all_
unsupported conditions at once.

It should make it easier to gauge how much work is neded to support a
certain firmware.

(cherry picked from commit e21721ae41aaf91a8ac9fd539bba4078770b33d0)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
(cherry picked from commit dae1cc891e9ef75fff34957e70747c50c188be79)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 src/thd_engine_adaptive.cpp | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index 8ca7e2b..fb33849 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -402,18 +402,23 @@ int cthd_engine_adaptive::verify_condition(struct condition condition) {
 	if (condition.condition == Default)
 		return 0;
 
-	thd_log_fatal("Unsupported condition %d, exiting\n", condition.condition);
+	thd_log_error("Unsupported condition %d\n", condition.condition);
 	return THD_ERROR;
 }
 
 int cthd_engine_adaptive::verify_conditions() {
+	int result = 0;
 	for (int i = 0; i < (int)conditions.size(); i++) {
 		for (int j = 0; j < (int)conditions[i].size(); j++) {
 			if (verify_condition(conditions[i][j]))
-				return THD_ERROR;
+				result = THD_ERROR;
 		}
 	}
-	return 0;
+
+	if (result != 0)
+		thd_log_error("Exiting due to unsupported conditions\n");
+
+	return result;
 }
 
 int cthd_engine_adaptive::evaluate_condition(struct condition condition) {
-- 
2.27.0

